### PR TITLE
Fix/wrong time in gll

### PR DESCRIPTION
**Pull request by msallin**
## Summary

- Extract shared `formatDatetime()` utility into `nmea.js` to consistently parse ISO 8601 datetime strings into NMEA0183 UTC time/date components
- Fix GLL sentence: was using local time (`getHours()`) instead of UTC (`getUTCHours()`) — this was the original bug reported in #96
- Refactor RMC, ZDA, and GGA to use the shared `formatDatetime()` instead of duplicated inline parsing
- Fix NMEA0183 time format: use actual centiseconds from input datetime instead of hardcoded `.020`
- ZDA now correctly outputs 4-digit year per spec

## Changes

- `nmea.js`: add `formatDatetime()` — parses ISO 8601 to `{ hours, minutes, seconds, centiseconds, time, day, month, year, date }` using UTC methods
- `sentences/GLL.js`: use `formatDatetime()` (fixes the UTC bug)
- `sentences/RMC.js`: use `formatDatetime()` (removes duplicated code)
- `sentences/GGA.js`: use `formatDatetime()` (removes duplicated code)
- `sentences/ZDA.js`: use `formatDatetime()` with full 4-digit year

## Test plan

- [x] Add `formatDatetime()` unit tests: UTC, timezone offsets (+02:00, -05:00, +00:00), fractional seconds, midnight, year boundary, day-crossing timezone, invalid inputs (number, null, undefined, empty string)
- [x] Add GLL sentence integration tests: UTC datetime, timezone offset (verifies the UTC fix), fractional seconds, null position
- [x] Add ZDA sentence integration tests: UTC datetime, timezone offset, day-crossing timezone, 4-digit year verification
- [x] Update GGA test expectations for new time format with centiseconds
- [x] All 28 tests passing

### CODE REVIEW

File: nmea.js [hunk 1]
```diff
@@ -140,6 +140,43 @@ function radsToPositiveDeg(r) {
...
+    const seconds = ('00' + datetime.getUTCSeconds()).slice(-2);
+
+    const day = ('00' + datetime.getUTCDate()).slice(-2);
+    const month = ('00' + (datetime.getUTCMonth() + 1)).slice(-2); // months from 1-12
+    const year = ('00' + datetime.getUTCFullYear()).slice(-2);
+    return {
+      hours: hours,
+      minutes: minutes,
+      seconds: seconds,
+      // NMEA0183 time format is hhmmss.ss
```
Comment (msallin) [inline nmea.js:158]: Im confused with the .ss; In ZDA '.020' is added as constant. Can you advice? The current PR doesn't change the behavior with that regard.

**Comment by msallin** [comment]
Re: the question about `.ss` in `hhmmss.ss`:

The `.ss` in the NMEA0183 time format represents **centiseconds** (hundredths of a second). Per the spec, the time field always includes two decimal places — e.g. `201530.00` means 20:15:30 and 00 centiseconds.

The hardcoded `.020` was a bug — it always output 02 centiseconds regardless of the actual input time. I've updated `formatDatetime()` to extract the real centiseconds from the input datetime's milliseconds (`Math.floor(ms / 10)`) and include them directly in the `time` field as `hhmmss.ss`. GLL and ZDA no longer append anything manually.

References:
- [SBG Systems NMEA Reference — ZDA example shows `201530.00`](https://developer.sbg-systems.com/sbgECom/5.1/nmea_msg.html)
- [Trimble NMEA-0183 GLL — format `hhmmss.ss`](https://receiverhelp.trimble.com/alloy-gnss/en-us/NMEA-0183messages_GLL.html)

File: nmea.js (hunk 1 above)
Comment (msallin) [reply nmea.js]: The `.ss` represents **centiseconds** (hundredths of a second). The NMEA0183 time format is `hhmmss.ss` — e.g. `201530.00` means 20:15:30.00.

The hardcoded `.020` was a bug — it always output 02 centiseconds regardless of the actual time. `formatDatetime()` now extracts real centiseconds from the input datetime's milliseconds (`Math.floor(ms / 10)`) and includes them in the `time` field. GLL and ZDA no longer append anything manually.

References:
- [SBG Systems NMEA Reference — ZDA example shows `201530.00`](https://developer.sbg-systems.com/sbgECom/5.1/nmea_msg.html)
- [Trimble NMEA-0183 GLL — format `hhmmss.ss`](https://receiverhelp.trimble.com/alloy-gnss/en-us/NMEA-0183messages_GLL.html)

**Comment by msallin** [comment]
@tkurki PR is open for a while. While not urgent, the refacor is prone to cause conflicts. Appreciate a review (and merge) when you find some time.
